mock/world/mockESDTAccount.go

```diff
@@ -149,8 +149,8 @@ func (a *Account) SetTokenData(tokenKey []byte, tokenData *esdt.ESDigitalToken)
 	if err != nil {
 		return err
 	}
-
-	return a.DataTrieTracker().SaveKeyValue(tokenKey, marshaledData)
+	a.Storage[string(tokenKey)] = marshaledData
```

Comment:
do we still need mockESDTAccount ? if yes, I think it was a little bit safer to use the SaveKeyValue as there is the Journal increased for ESDT transfers, even in mocks.

Reply:
`SaveKeyValue` ended up being called before the test was even started (during test setup). This led to snapshots being created before the test, causing Mandos rollbacks to have strange behavior and tests were failing for no apparent reason.

Avoiding `SaveKeyValue` is safer here, because the functions in `mockESDTAccount.go` are only called during test setup, when they must leave the journal unaffected.